### PR TITLE
Adding documentation related to raven 5 persistence for Audit

### PR DESCRIPTION
**Pull request by WojcikMike**
Added the following pages:
 - persistence page for SC.Audit
 - Modified the page defining expiration

When we agree on the wording we should create a page describing persistence for the error instance

### CODE REVIEW

File: servicecontrol/audit-instances/persistence.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: ServiceControl Audit persistence
+summary: Information about ServiceControl Audit persistence options.
+reviewed: 2022-10-17
+---
+
+ServiceControl audit instances store their data in an RavenDB embedded database. In the ServiceControl.Audit version 4.26 and up, new instances store their data in RavenDB version 5. Instances less than version 4.26 use RavenDB version 3.5.
```
Comment (mikeminutillo) [inline servicecontrol/audit-instances/persistence.md:7]: Using [version language guidance](https://github.com/particular/docs.particular.net#version-language)

```suggestion
ServiceControl audit instances store their data in a RavenDB embedded database. In ServiceControl.Audit version 4.26 and above, new instances store their data in RavenDB version 5. Instances created by version 4.25 and below use RavenDB version 3.5.
```

File: servicecontrol/audit-instances/persistence.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+Upgrading ServiceControl.Audit instances into version 4.26 or higher does not change the database version. So instance that uses RavenDB version 3.5 upgraded to the newest version will still be using RavenDB version 3.5.
```
Comment (mikeminutillo) [inline servicecontrol/audit-instances/persistence.md:9]:
```suggestion
Upgrading ServiceControl.Audit instances into version 4.26 or higher does not change the database version. So instance that use RavenDB version 3.5 upgraded to the newest version will still be using RavenDB version 3.5.
```

File: servicecontrol/how-purge-expired-data.md [hunk 3]
```diff
@@ -4,11 +4,17 @@ summary: Configuring ServiceControl's data retention policy
...
 
 ServiceControl stores audit and error data. Any audit and error data that is older than the specified thresholds is deleted from the embedded RavenDB. The expiration thresholds for both faulted and audited messages must be set during installation. These values can be modified later by launching ServiceControl Management and editing the configuration settings for the instance.
 
 Note: The expiration process curates only the data in the embedded RavenDB. Audit and error forwarding queues are not curated or managed by ServiceControl. To turn these settings off, launch ServiceControl Management and edit the configuration settings for the instance.
 
 Warning: The database will not automatically shrink in size after reducing the retention period. Ensure ServiceControl had time to purge all expired messages and then [Compact the database](db-compaction.md).
+
+## Differences in purge implementations
+
+Expiration of error and audit data is implemented by runnig a reccuring task that checks the documents for old data and deleting the rows that should be deleted. Changing the expiration setting results in it being picked up during the next time the service is run.
```
Comment (mikeminutillo) [inline servicecontrol/how-purge-expired-data.md:19]:
```suggestion
Expiration of error and audit data is implemented by a reccuring task that checks the documents for old data and deletes documents that have expired. Changing the expiration setting results in it being picked up during the next time the service is run. 
```

File: servicecontrol/how-purge-expired-data.md [hunk 4]
```diff
@@ -4,11 +4,17 @@ summary: Configuring ServiceControl's data retention policy
...
 
 Note: The expiration process curates only the data in the embedded RavenDB. Audit and error forwarding queues are not curated or managed by ServiceControl. To turn these settings off, launch ServiceControl Management and edit the configuration settings for the instance.
 
 Warning: The database will not automatically shrink in size after reducing the retention period. Ensure ServiceControl had time to purge all expired messages and then [Compact the database](db-compaction.md).
+
+## Differences in purge implementations
+
+Expiration of error and audit data is implemented by runnig a reccuring task that checks the documents for old data and deleting the rows that should be deleted. Changing the expiration setting results in it being picked up during the next time the service is run. 
+In the ServiceControl.Audit version 4.26 and up expiration is done using the persistance mechanisms, that results in metadata key to be stored with every message. Thanks to that, purging of the data can be handled fully by database itself. The drawback is that changing of expiration setting is only applied to new rows.
```
Comment (mikeminutillo) [inline servicecontrol/how-purge-expired-data.md:20]: Using version language guidance.

```suggestion
In ServiceControl.Audit version 4.26 and above expiration is handled by the database automatically. Each audited message contains a metadata key that tells the database when to remove it. When the expiration setting changes, the new value is only applied to new audit data. Existing audit meta is not changed. Messages already in the database are not affected by the change to the expiration setting.
```

File: servicecontrol/how-purge-expired-data.md [hunk 5]
```diff
@@ -4,11 +4,17 @@ summary: Configuring ServiceControl's data retention policy
...
+reviewed: 2022-10-18
 ---
 
 ServiceControl stores audit and error data. Any audit and error data that is older than the specified thresholds is deleted from the embedded RavenDB. The expiration thresholds for both faulted and audited messages must be set during installation. These values can be modified later by launching ServiceControl Management and editing the configuration settings for the instance.
 
 Note: The expiration process curates only the data in the embedded RavenDB. Audit and error forwarding queues are not curated or managed by ServiceControl. To turn these settings off, launch ServiceControl Management and edit the configuration settings for the instance.
 
 Warning: The database will not automatically shrink in size after reducing the retention period. Ensure ServiceControl had time to purge all expired messages and then [Compact the database](db-compaction.md).
+
+## Differences in purge implementations
```
Comment (andreasohlund) [inline servicecontrol/how-purge-expired-data.md:17]:
```suggestion
## Differences in message rentention implementations
```

File: servicecontrol/how-purge-expired-data.md [hunk 6]
```diff
@@ -4,11 +4,17 @@ summary: Configuring ServiceControl's data retention policy
...
+
+Expiration of error and audit data is implemented by a reccuring task that checks the documents for old data and deletes documents that have expired. Changing the expiration setting results in it being picked up during the next time the service is run.
```
Comment (mauroservienti) [inline servicecontrol/how-purge-expired-data.md:19]:
```suggestion
The expiration of error and audit data is implemented by a recurring task that checks and deletes expired documents. Changing the expiration setting results in it being picked up the next time the service is run. 
```

File: servicecontrol/how-purge-expired-data.md [hunk 7]
```diff
@@ -4,11 +4,17 @@ summary: Configuring ServiceControl's data retention policy
...
 
 Note: The expiration process curates only the data in the embedded RavenDB. Audit and error forwarding queues are not curated or managed by ServiceControl. To turn these settings off, launch ServiceControl Management and edit the configuration settings for the instance.
 
 Warning: The database will not automatically shrink in size after reducing the retention period. Ensure ServiceControl had time to purge all expired messages and then [Compact the database](db-compaction.md).
+
+## Differences in purge implementations
+
+Expiration of error and audit data is implemented by a reccuring task that checks the documents for old data and deletes documents that have expired. Changing the expiration setting results in it being picked up during the next time the service is run. 
+In ServiceControl.Audit version 4.26 and above expiration is handled by the database automatically. Each audited message contains a metadata key that tells the database when to remove it. When the expiration setting changes, the new value is only applied to new audit data. Existing audit meta is not changed. Messages already in the database are not affected by the change to the expiration setting.
```
Comment (mauroservienti) [inline servicecontrol/how-purge-expired-data.md:20]:
```suggestion
In ServiceControl.Audit version 4.26 and above expiration is handled by the database automatically. Each audited message contains a metadata key that tells the database when to remove it. When the expiration setting changes, the new value is only applied to new audit data. Metadata for already ingested messages is not changed. Only new audited messages are affected by the change to the expiration setting.
```

File: servicecontrol/audit-instances/persistence.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+title: ServiceControl Audit persistence
+summary: Information about ServiceControl Audit persistence options.
+reviewed: 2022-10-17
+---
+
+ServiceControl audit instances store their data in a RavenDB embedded database. In ServiceControl.Audit version 4.26 and above, new instances store their data in RavenDB version 5. Instances created by version 4.25 and below use RavenDB version 3.5.
+
+Upgrading ServiceControl.Audit instances into version 4.26 or higher does not change the database version. So instance that use RavenDB version 3.5 upgraded to the newest version will still be using RavenDB version 3.5.
```
Comment (mauroservienti) [inline servicecontrol/audit-instances/persistence.md:9]:
```suggestion
Upgrading ServiceControl.Audit instances to version 4.26 or higher does not change the database version. Instances using RavenDB version 3.5, when upgraded to the newest version, will still be using RavenDB version 3.5.
```

**Review by mauroservienti — CHANGES_REQUESTED** [review]

**Comment by WojcikMike** [comment]
We probably should wait with this PR until we actually release SC right?

**Review by mauroservienti — APPROVED** [review]

File: servicecontrol/audit-instances/persistence.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+---
+title: ServiceControl Audit persistence
+summary: Information about ServiceControl Audit persistence options.
+reviewed: 2022-10-17
+---
+
+ServiceControl audit instances store their data in a RavenDB embedded database. In ServiceControl.Audit version 4.26 and above, new instances store their data in RavenDB version 5. Instances created by version 4.25 and below use RavenDB version 3.5.
```
Comment (andreasohlund) [inline servicecontrol/audit-instances/persistence.md:7]:
```suggestion
ServiceControl audit instances store their data in a RavenDB embedded database. In ServiceControl.Audit version 4.26 and above new instances use RavenDB version 5. Instances created by version 4.25 and below use RavenDB version 3.5.
```

File: servicecontrol/audit-instances/persistence.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+
+Upgrading ServiceControl.Audit instances to version 4.26 or higher does not change the database version. Instances using RavenDB version 3.5, when upgraded to the newest version, will still be using RavenDB version 3.5.
```
Comment (andreasohlund) [inline servicecontrol/audit-instances/persistence.md:9]: We should link to the upgrade guide that @mikeminutillo is working on

File: servicecontrol/persistence.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
+---
+title: ServiceControl persistence
+summary: Information about ServiceControl persistence.
+reviewed: 2022-10-17
+---
+
+ServiceControl error instances store their data in a RavenDB embedded database. All instances use RavenDB version 3.5.
+
+WARNING: The ServiceControl Audit RavenDB embedded instance is used exclusively by ServiceControl and is not intended for external manipulation or modifications.
```
Comment (andreasohlund) [inline servicecontrol/persistence.md:9]:
```suggestion
WARNING: The ServiceControl RavenDB embedded database is used exclusively by ServiceControl and is not intended for external manipulation or modifications.
```

File: servicecontrol/audit-instances/persistence.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+WARNING: The ServiceControl Audit RavenDB embedded instance is used exclusively by ServiceControl and is not intended for external manipulation or modifications.
```
Comment (andreasohlund) [inline servicecontrol/audit-instances/persistence.md:11]:
```suggestion
WARNING: The ServiceControl Audit RavenDB embedded database is used exclusively by ServiceControl and is not intended for external manipulation or modifications.
```

**Pull request merged**